prow/cmd/crier/main_test.go

```diff
@@ -60,17 +72,65 @@ func TestOptions(t *testing.T) {
 					"foo": {"bar"},
 				},
 				configPath: "foo",
+				github:     defaultGitHubOptions,
+			},
+		},
+		//PubSub Reporter
+		{
+			name: "pubsub workers, sets workers",
+			args: []string{"--pubsub-workers=7", "--config-path=baz"},
+			expected: &options{
+				pubsubWorkers:  7,
+				configPath:     "baz",
+				github:         defaultGitHubOptions,
+				gerritProjects: defaultGerritProjects,
+			},
+		},
+		{
+			name: "pubsub workers set to negative, rejects",
+			args: []string{"--pubsub-workers=-3", "--config-path=foo"},
+		},
+		//Slack Reporter
+		{
+			name: "slack workers, sets workers",
+			args: []string{"--slack-workers=13", "--slack-token-file=/bar/baz", "--config-path=foo"},
+			expected: &options{
+				slackWorkers:   13,
+				slackTokenFile: "/bar/baz",
+				configPath:     "foo",
+				github:         defaultGitHubOptions,
+				gerritProjects: defaultGerritProjects,
+			},
+		},
+		{
+			name: "slack missing --slack-token, rejects",
+			args: []string{"--slack-workers=1", "--config-path=foo"},
+		},
+		{
+			name: "slack with --dry-run, sets",
+			args: []string{"--slack-workers=13", "--slack-token-file=/bar/baz", "--config-path=foo", "--dry-run", "--deck-url=http://www.example.com"},
+			expected: &options{
+				slackWorkers:   13,
+				slackTokenFile: "/bar/baz",
+				configPath:     "foo",
+				dryrun:         true,
+				client: prowflagutil.ExperimentalKubernetesOptions{
+					DeckURI: "http://www.example.com",
+				},
+				github:         defaultGitHubOptions,
+				gerritProjects: defaultGerritProjects,
 			},
 		},
+		{
+			name: "Dry run with no --deck-url, rejects",
+			args: []string{"--slack-workers=13", "--slack-token-file=/bar/baz", "--config-path=foo", "--dry-run"},
+		},
 	}
 
 	for _, tc := range cases {
 		flags := flag.NewFlagSet(tc.name, flag.ContinueOnError)
-		actual := options{
-			gerritProjects: gerritclient.ProjectsFlag{},
-		}
+		actual := options{}
```

Comment:
nit: prefer `var actual options` when do not initialize the value.

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_use_a_consistent_declaration_style

Reply:
Valid nits. I've fixed these, and a couple of other initialization points that seemed either unclear or non-idiomatic